File: server/FitMate.Core/JsonModels/Subscriptions/SubscriptionLimitErrorModel.cs
using FitMate.DB.Enums;

namespace FitMate.Core.JsonModels.Subscriptions;

public class SubscriptionLimitErrorModel
{
    public string Code { get; set; } = "subscription_limit_reached";
    public SubscriptionFeature Feature { get; set; }
    public int? Limit { get; set; }
    public int Used { get; set; }
    public int Reserved { get; set; }
    public DateTime? ResetsAt { get; set; }
    public bool UpgradeAvailable { get; set; }
}

File: client/src/types/JsonModels/Subscriptions/SubscriptionLimitErrorModel.ts
import type { SubscriptionFeature } from "../Enums/SubscriptionFeature";

export interface SubscriptionLimitErrorModel
{
	code: string;
	feature: SubscriptionFeature;
	limit?: number;
	used: number;
	reserved: number;
	resetsAt?: string;
	upgradeAvailable: boolean;
}
